pages/api/savedInsulinData/[id].js

```diff
@@ -0,0 +1,20 @@
+import { getCategoryById } from '../../../services/insulinAppDataService';
+import SavedInsulinData from '../../../models/InsulinApp';
+
+export default async function handler(request, response) {
+  const { id } = request.query;
+  switch (request.method) {
+    case 'GET':
+      const cardData = getCategoryById(id);
```

Comment:
the naming seems a bit off. You store the results of a function called `getCategoryById()` in a variable called `cardData`. This is confusing.